spec/cordova/platform/getPlatformDetailsFromDir.spec.js

```diff
@@ -0,0 +1,79 @@
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var path = require('path');
+var fs = require('fs');
+var Q = require('q');
+var rewire = require('rewire');
+var cordova_util = require('../../../src/cordova/util');
+var platform_getPlatformDetails = rewire('../../../src/cordova/platform/getPlatformDetailsFromDir');
+var events = require('cordova-common').events;
+var fail;
+
+describe('cordova/platform/getPlatformDetailsFromDir', function () {
+    var package_json_mock;
+    package_json_mock = jasmine.createSpyObj('package json mock', ['cordova', 'dependencies']);
+    package_json_mock.name = 'io.cordova.hellocordova';
+    package_json_mock.version = '1.0.0';
+
+    beforeEach(function () {
+        spyOn(Q, 'reject');
+        spyOn(fs, 'existsSync');
+        spyOn(cordova_util, 'requireNoCache');
+        spyOn(events, 'emit');
+    });
+
+    it('should throw if no config.xml or pkgJson', function (done) {
+        platform_getPlatformDetails('dir', ['ios']);
+        expect(Q.reject).toHaveBeenCalledWith(jasmine.stringMatching(/does not seem to contain a valid package.json or a valid Cordova platform/));
+        done();
+    });
+
+    it('should throw if no platform is provided', function (done) {
+        cordova_util.requireNoCache.and.returnValue({});
+        platform_getPlatformDetails('dir');
+        expect(Q.reject).toHaveBeenCalledWith(jasmine.stringMatching(/does not seem to contain a Cordova platform:/));
+        done();
+    });
+
+    it('should return a promise with platform and version', function (done) {
+        fs.existsSync.and.callFake(function(filePath) {
+            if(path.basename(filePath) === 'package.json') {
+                return true;
+            } else {
+                return false;
+            }
+        });
+        cordova_util.requireNoCache.and.returnValue(package_json_mock);
+        platform_getPlatformDetails('dir', ['cordova-android'])
+        .then(function(result) {
+            expect(result.platform).toBe('io.cordova.hellocordova');
+            expect(result.version).toBe('1.0.0');
+            expect(Q.reject).not.toHaveBeenCalled();
+        }).fail(function (err) {
+            fail('unexpected failure handler invoked!');
+            console.error(err);
+        }).done(done);
+    });
+
+    it('should remove the cordova- prefix from the platform name for known platforms', function (done) {
+        platform_getPlatformDetails.platformFromName('cordova-ios');
+        expect(events.emit).toHaveBeenCalledWith('verbose', jasmine.stringMatching(/Removing "cordova-" prefix/));
+        expect(platform_getPlatformDetails.platformFromName('cordova-ios')).toBe('ios');
```

Comment:
You can take the expect from line 76 and combine it with line 74 and delete 76. So line 74 gets replaced by line 76. Line 75 should still pass